builtin/credential/ldap/path_config.go

```diff
@@ -143,8 +166,22 @@ func (b *backend) pathConfigWrite(
 	if groupdn != "" {
 		cfg.GroupDN = groupdn
 	}
+	groupfilter := d.Get("groupfilter").(string)
```

Comment:
OK, this gets a bit tricky. We want the user to be able to specify an empty groupfilter in order to force only local configuration to be queried. However, as mentioned earlier, we also want to handle upgrades safely, which means using the default filter (equivalent to the previous logic) if nothing is specified.

So, if we only have logic here to check this condition, we fail upgrades until they rewrite their configuration; if we have logic elsewhere we can't actually change it here.

`groupattr` also has this problem.

The only idea I have right now would be to make these members of the configuration struct `*string` rather than `string`. That way, a nil pointer indicates that nothing was set in the configuration, whereas a valid pointer means that the value pointed to (even if `""`) was set by the user. In that case, you'd change the logic here to use the multi-valued `d.GetOk` which tells you whether something was set or not; if so, you can store the string value, and if not, you can keep the configuration value nil.

Any better ideas?